### PR TITLE
[14.0][IMP] contract: Remove deprecated domain in onchange

### PR DESCRIPTION
**Pull request by rousseldenis**

### CODE REVIEW

**Comment by rousseldenis** [comment]
Fixes:

image

**Comment by pedrobaeza** [comment]
But there should be a replacement through a static domain or something.

**Comment by pedrobaeza** [comment]
Maybe through `=?` operator.

**Comment by rousseldenis** [comment]
> But there should be a replacement through a static domain or something.

But correct me if I'm wrong, the current behaviour is ineffective. The domain return has no influence.

So, this can be merged and another PR should be done to get the domain

**Comment by pedrobaeza** [comment]
Being deprecated doesn't mean the domain is not working. It should, as this message is a warning for being replacing it without still removing the support.

**Comment by rousseldenis** [comment]
> Being deprecated doesn't mean the domain is not working. It should, as this message is a warning for being replacing it without still removing the support.

Indeed, I'll modify this accordingly

File: contract/tests/test_contract.py [hunk 1]
```diff
@@ -483,22 +504,35 @@ def test_onchange_partner_id(self):
             self.contract.partner_id.property_product_pricelist,
         )
 
+    def test_onchange_partner_id_domain(self):
+        with Form(self.contract) as contract_form:
+            contract_form.invoice_partner_id = self.partner_2
+        self.assertNotEqual(
```
Comment (rousseldenis) [inline contract/tests/test_contract.py]: @pedrobaeza Both changes done and functional in views.

The first test for uom is ok, but this one does not seem to work. Do you have an idea?

**Review by pedrobaeza — APPROVED** [review]

File: contract/tests/test_contract.py (hunk 1 above)
Comment (pedrobaeza) [reply contract/tests/test_contract.py]: Are you trying to test domains? I think that through direct assignation, there's no domain restriction applied.

Comment (rousseldenis) [reply contract/tests/test_contract.py]: In uom the behaviour is working in test. If you try to assign a wrong uom, it bypasses it.

Comment (rousseldenis) [reply contract/tests/test_contract.py]: @pedrobaeza Or do you know the property of a field that has the evaluated domain value ?

Comment (pedrobaeza) [reply contract/tests/test_contract.py]: I have never tested such domains being frank. The only way I know right now is to call fields_view_get and check the domain attribute in the resulting xml. I can investigate later.

Comment (rousseldenis) [reply contract/tests/test_contract.py]: Ok, I have found the way to test it. Set the invoice_partner_id to void before

Comment (rousseldenis) [reply contract/tests/test_contract.py]: So, this one is ready

Comment (pedrobaeza) [reply contract/tests/test_contract.py]: Great, TIL about this way of testing domains through Form

**Comment by rousseldenis** [comment]
Don't find the way to test that (Form() does not take into account field domain not defined on views). But at least the domains are working

**Comment by rousseldenis** [comment]
> Don't find the way to test that (Form() does not take into account field domain not defined on views). But at least the domains are working

Tested the domain result in view as suggested by @pedrobaeza

**Comment by pedrobaeza** [comment]
OK, don't worry. Not sure if the `fields_view_get` thing is adding enough value to the tests, and they may be conflicting even if other customizations are done. I would remove them. They get meaning when they are dynamic according anything, and you test that they get the proper dynamic value. I will investigate if there's a way to test such domains.

**Review by hparfr — APPROVED** [review]

**Pull request merged**